keel-web/src/test/kotlin/com/netflix/spinnaker/keel/rest/AuthorizationSupportTests.kt

```diff
@@ -0,0 +1,236 @@
+import com.netflix.spinnaker.keel.api.DeliveryConfig
+import com.netflix.spinnaker.keel.api.Environment
+import com.netflix.spinnaker.keel.api.id
+import com.netflix.spinnaker.keel.persistence.CombinedRepository
+import com.netflix.spinnaker.keel.rest.AuthorizationSupport.Action
+import com.netflix.spinnaker.keel.rest.AuthorizationSupport.TargetEntity
+import com.netflix.spinnaker.keel.test.locatableResource
+import com.netflix.spinnaker.kork.dynamicconfig.DynamicConfigService
+import dev.minutest.junit.JUnit5Minutests
+import dev.minutest.rootContext
+import io.mockk.clearAllMocks
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.mockkStatic
+import org.springframework.security.access.AccessDeniedException
+import org.springframework.security.core.Authentication
+import org.springframework.security.core.context.SecurityContextHolder
+import strikt.api.expectCatching
+import strikt.api.expectThat
+import strikt.api.expectThrows
+import strikt.assertions.isFalse
+import strikt.assertions.isTrue
+import strikt.assertions.succeeded
+
+internal class AuthorizationSupportTests : JUnit5Minutests {
+  private val dynamicConfigService: DynamicConfigService = mockk(relaxed = true)
+  private val permissionEvaluator: FiatPermissionEvaluator = mockk(relaxed = true)
+  private val combinedRepository: CombinedRepository = mockk(relaxed = true)
+  private val resource = locatableResource()
+  private val deliveryConfig = DeliveryConfig(
+    name = "manifest",
+    application = ApplicationControllerTests.application,
+    serviceAccount = "keel@spinnaker",
+    artifacts = emptySet(),
+    environments = setOf(Environment("test", setOf(resource)))
+  )
+  private val application = deliveryConfig.application
+
+  fun tests() = rootContext<AuthorizationSupport> {
+    fixture {
+      AuthorizationSupport(permissionEvaluator, combinedRepository, dynamicConfigService)
+    }
+
+    after {
+      clearAllMocks()
+    }
+
+    context("authorization is enabled") {
+      before {
+        mockkStatic("org.springframework.security.core.context.SecurityContextHolder")
+
+        every {
+          SecurityContextHolder.getContext()
+        } returns mockk(relaxed = true)
+
+        every {
+          dynamicConfigService.isEnabled("keel.authorization", true)
+        } returns true
+
+        every {
+          combinedRepository.getResource(resource.id)
+        } returns resource
+
+        every {
+          combinedRepository.getDeliveryConfig(deliveryConfig.name)
+        } returns deliveryConfig
+
+        every {
+          combinedRepository.getDeliveryConfigForApplication(application)
+        } returns deliveryConfig
+      }
+
+      listOf(Action.READ, Action.WRITE).forEach { action ->
+        context("user has no ${action.name} access to application") {
+          before {
+            every {
+              dynamicConfigService.isEnabled("keel.authorization", true)
+            } returns true
```

Comment:
Why is it necessary to re-do this when it's done in the parent `before`?

Reply:
Good catch. Will remove.